tests/testthat/test-environment.R

```diff
@@ -41,3 +41,35 @@ for (x in names(strategy_types)) {
     expect_identical(sapply(hmid, env$canopy$canopy_interpolator$eval), sapply(hmid, interplator$eval))
   })
 }
+
+test_that("FF16 rainfall spline", {
+  context(sprintf("Rainfall-FF16",x))
+  env <- make_environment("FF16")
+  
+  ## simple quadratic
+  x <- seq(-10, 10, 2)
+  y <- x^2
+  env$rainfall_init(x, y)
+  
+  ## discrete points used for spline creation
+  expect_equal(env$rainfall_eval(2), 4)
+  expect_equal(env$rainfall_eval(-2), 4)
+  
+  ## interpolated points
+  expect_equal(env$rainfall_eval(3), 9, tolerance=1e-2)
```

Comment:
tolerance of 0.01 is very generous. I would have expected it to be accurate at much higher resolution. Does it not pass with default tolerance ?

Similar everywhere tolerance is used.

Reply:
On that line specifically I was able to pass with `0.001 (1e-3)` tolerance at most - however, if I create the spline with more control points by using a smaller step value:
```R
x <- seq(-10, 10, 0.41) # as opposed to step of 2
y <- x^2
env$rainfall_init(x, y)
```
then the tests pass with a much smaller tolerance of `1e-7` or `1e-6`, so I think its a limitation of the spline when given few control points. Ill change the tests to use more control points.